src/extension/options-page/DashboardRouters/Contacts.tsx

```diff
@@ -102,6 +104,28 @@ export default function DashboardContactsRouter() {
         [size, setSize],
     )
 
+    const contactUpdateHandler = (mutatedContact: MutatedContact) => {
+        const mutatedData = data?.map((contacts) =>
+            contacts.map((contact) => {
+                return isEqual(contact.identifier, mutatedContact.identifier)
+                    ? { ...contact, ...mutatedContact }
+                    : contact
+            }),
+        )
+
+        mutate(mutatedData, false)
+        revalidate()
```

Comment:
Instead of using `mutate(mutatedData, true)` which does not work. Split the progress as:

1. mutate without revalidation
2. call revalidate manually